src/java/fr/paris/lutece/plugins/document/modules/metadatadublincore/business/DublinCoreMetadata.java

```diff
@@ -475,32 +475,45 @@ public String getXml(  )
      */
     public void load( String strXmlData )
     {
-        // Configure Digester from XML ruleset
-        URL rules = getClass(  ).getResource( FILE_RULES );
-        Digester digester = DigesterLoader.createDigester( rules );
-
-        // Push empty List onto Digester's Stack
-        digester.push( this );
-
-        try
-        {
-            if ( strXmlData != null )
-            {
-                StringReader sr = new StringReader( strXmlData );
-                digester.parse( sr );
-            }
-        }
-        catch ( FileNotFoundException e )
-        {
-            AppLogService.error( e.getMessage(  ), e );
-        }
-        catch ( SAXException e )
-        {
-            AppLogService.error( e.getMessage(  ), e );
-        }
-        catch ( IOException e )
-        {
-            AppLogService.error( e.getMessage(  ), e );
-        }
+    	 if (strXmlData == null || strXmlData.isEmpty()) {
+    	        return;
+    	    }
+
+    	    URL rules = getClass().getResource(FILE_RULES);
+    	    if (rules == null) {
+    	        AppLogService.error("Cannot find Digester rules file: " + FILE_RULES);
+    	        return;
+    	    }
+    	    
+    	    try
+    	    {
+	    	    DigesterLoader loader = DigesterLoader.newLoader(new FromXmlRulesModule() {
+	    	        @Override
+	    	        protected void loadRules() {
+	    	            loadXMLRules(rules);
+	    	        }
+	    	    });
+    	    
+	    	    try
+	    	    {
+		    	    Digester digester = loader.newDigester();
+		    	    digester.push(this);
+		
+		    	    try (StringReader sr = new StringReader(strXmlData)) {
+		    	        digester.parse(sr);
+		    	    } 
+		    	    catch (IOException | SAXException e) {
+		    	        AppLogService.error("Error parsing DublinCore XML: " + e.getMessage(), e);
```

Comment:
Use the recommended syntax:
AppLogService.error("Error parsing DublinCore XML: {} ",  e.getMessage(), e);